lib/stats.js

```diff
@@ -116,6 +116,22 @@ class Stats {
         this._checkFinished(fulfill);
     }
 
+    _Network_requestWillBeSentExtraInfo(fulfill, reject, params) {
+        const {requestId, headers: newHeaders} = params;
+        const entry = this.entries.get(requestId);
+        if (!entry) {
+            return;
+        }
+        Object.keys(newHeaders).forEach(newHeader => {
+            const existingHeader = Object.keys(entry.requestParams.request.headers).filter(existingHeader => existingHeader.toLowerCase() === newHeader.toLowerCase());
+            if (existingHeader.length > 0) {
+                entry.requestParams.request.headers[existingHeader] = newHeaders[newHeader];
```

Comment:
A thought here. I was originally going to suggest switching to Array::find, but that would give you just the first occurrence of that header in the original request headers collection. 
Then I looked at your approach, and what happens in the following use case:
entry.requestParams has more than one entry for the "newHeader"?
we would be only replacing the first instance of the new header in the old entries right?
given `entry.requestParams.request.headers[index]`.

do we care? if we do not, then Array::find will give you the first entry that matches the predicate, if needed ::findIndex could be used. It will save you the iterations of going through every header to see if the predicate matches when we do ::filter. 
I think this will work for all headers that are present only once in the collection (unique by name), but it won't for the others given that we are not modifying the filtered collection.

Reply:
I muddled over the idea of overwriting vs simply appending.  I wish I know more about the contract of this other event.  Will it never provide a duplicate header?   But on further consideration, perhaps what we need to do here is check if an exact header exists (same key (ignore case) and same value(case-sensitive).  if it does, don't append it.  If not, append.  I'll rework this logic.